### PR TITLE
Add new os_keystone_role module.

### PR DESCRIPTION
**Pull request by Shrews**
This new module allows for creating and deleting Keystone
roles.

### CODE REVIEW

**Comment by Shrews** [comment]
ping @emonty @j2sol

**Comment by gregdek** [comment]
Thanks @Shrews for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

File: cloud/openstack/os_keystone_role.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    type: dictionary
+    contains:
+        id:
+            description: Unique role ID.
+            type: string
+            sample: "677bfab34c844a01b88a217aa12ec4c2"
+        name:
+            description: Role name.
+            type: string
+            sample: "demo"
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_role.py:75]: I'm getting back `HUMAN_ID` and `NAME_ATTR` and `human_id` from my cloud, as well as a `links` hash.

```
<localhost> REMOTE_MODULE os_keystone_role cloud=trustytest name=jlkderp
localhost | success >> {
    "changed": false,
    "role": {
        "HUMAN_ID": false,
        "NAME_ATTR": "name",
        "human_id": null,
        "id": "27d4a926fa5a4d69ab0d3709ebaf204a",
        "links": {
            "self": "https://trustytest.openstack.blueboxgrid.com:5001/v3/roles/27d4a926fa5a4d69ab0d3709ebaf204a"
        },
        "name": "jlkderp"
    }
}
```

Not different enough to block this PR, just informational data.

**Comment by omgjlk** [comment]
Tested on one of my clouds.

shipit

**Comment by emonty** [comment]
shipit

**Pull request merged**